test/parallel/rolldown.withdraw.merkleRoot.test.ts

```diff
@@ -0,0 +1,472 @@
+    minToBeSequencer.addn(1000),
+    chain,
+  );
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintNative(testUser),
+    Assets.mintNative(testUser2),
+    Sudo.sudoAs(testUser, stakeAndJoinExtrinsic),
+    Sudo.sudoAs(testUser2, stakeAndJoinExtrinsic),
+  );
+  await createAnUpdateAndCancelIt(
+    testUser,
+    testUser2.keyRingPair.address,
+    chain,
+  );
+  const event = await Rolldown.waitForNextBatchCreated(
+    "Ethereum",
+    waitingBatchPeriod,
+  );
+  expect(event.source).toEqual("PeriodReached");
+  const l2Request = await Rolldown.getL2Request(
+    event.range.to.toNumber(),
+    chain,
+  );
+  expect(l2Request.cancel.requestId.id).toEqual(event.range.to.toNumber());
+  expect(l2Request.cancel.updater).toEqual(testUser.keyRingPair.address);
+  expect(l2Request.cancel.canceler).toEqual(testUser2.keyRingPair.address);
+});
+
+test("Given a withdraw extrinsic run WHEN token address does not exist THEN a withdraw fail and no l2update is stored", async () => {
+  const nonExistedToken = await getNonExistedTokenId();
+  const events = await signTx(
+    getApi(),
+    await Withdraw(
+      testUser,
+      10,
+      nonExistedToken.toHuman().ethereum,
+      "Ethereum",
+    ),
+    testUser.keyRingPair,
+  );
+  const filteredEvent = filterEventData(events, "WithdrawalRequestCreated");
+  expect(filteredEvent[0]).toBeUndefined();
+  const res = getEventResultFromMangataTx(events);
+  expect(res.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+  expect(res.data).toEqual("TokenDoesNotExist");
+});
+
+test("Given a withdraw extrinsic run WHEN token address exist but requester does not have funds THEN a withdraw fail and no l2update is stored", async () => {
+  await signTx(
+    getApi(),
+    await Withdraw(testUser, 10, ethIdL1Asset.ethereum, "Ethereum"),
+    testUser.keyRingPair,
+  ).then((events) => {
+    const res = getEventResultFromMangataTx(events);
+    expect(res.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+    expect(res.data).toEqual("NotEnoughAssets");
+  });
+});
+
+test("Given a succeed withdraw, when block is executed Then tokens are removed from user wallet", async () => {
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(ETH_ASSET_ID, testUser, BN_MILLION),
+  );
+  await testUser.refreshAmounts(AssetWallet.BEFORE);
+  await signTx(
+    getApi(),
+    await Withdraw(testUser, BN_HUNDRED, ethIdL1Asset.ethereum, "Ethereum"),
+    testUser.keyRingPair,
+  ).then((events) => {
+    const res = getEventResultFromMangataTx(events);
+    expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  });
+  await Rolldown.waitForNextBatchCreated("Ethereum", waitingBatchPeriod);
+  await testUser.refreshAmounts(AssetWallet.AFTER);
+  const withdrawalAmount = testUser
+    .getAsset(ETH_ASSET_ID)!
+    .amountBefore.free.sub(testUser.getAsset(ETH_ASSET_ID)!.amountAfter.free);
+  expect(withdrawalAmount).bnEqual(BN_HUNDRED);
```

Comment:
The test is about "block executed", so right after the the Withdraw is executed, the user tokens balance is affected. Not after the batch creation. 
Plz, move this to L185 or L186, right after the Tx execution and Before the batch.

Reply:
Done